### PR TITLE
Codecentric Spring Boot Admin support

### PR DESCRIPTION
**Pull request by joshlong**
adds checkboxes for Spring Boot Admin (UI, Server, Client)

### CODE REVIEW

File: initializr-service/src/main/resources/application.yml [hunk 1]
```diff
@@ -1252,6 +1252,33 @@ initializr:
           id: remote-shell
           description: CRaSH shell integration
           versionRange: "[1.0.0.RELEASE,1.5.0.RC1)"
+        - name: Codecentric Spring Boot Admin Server
+          description: Provides the Spring Boot Admin Server API (which can be paired with the UI)
+          id: spring-boot-admin-server
```
Comment (snicoll) [inline initializr-service/src/main/resources/application.yml]: Unfortunately, as [described in the guidelines for 3rd party starters](https://github.com/spring-io/initializr/wiki/Guidelines-for-3rd-Party-Starters), this naming is [not compliant](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-custom-starter-naming).

File: initializr-service/src/main/resources/application.yml [hunk 2]
```diff
@@ -1252,6 +1252,33 @@ initializr:
...
+          artifactId: spring-boot-admin-server
+          version: 1.5.6
+          groupId: de.codecentric
+          links:
+            - rel: reference
+              href: http://codecentric.github.io/spring-boot-admin/1.5.6/#getting-started
```
Comment (snicoll) [inline initializr-service/src/main/resources/application.yml]: That version is going  to be  annoying to upgrade. A reference to `current` would be interesting.

File: initializr-service/src/main/resources/application.yml [hunk 3]
```diff
@@ -1252,6 +1252,33 @@ initializr:
           id: remote-shell
           description: CRaSH shell integration
           versionRange: "[1.0.0.RELEASE,1.5.0.RC1)"
+        - name: Codecentric Spring Boot Admin Server
```
Comment (snicoll) [inline initializr-service/src/main/resources/application.yml]: I think a `versionRange` is in order here (unless there is support for Spring Boot 2 already, in which case I expect a separate version).

File: initializr-service/src/main/resources/application.yml (hunk 2 above)
Comment (joshiste) [reply initializr-service/src/main/resources/application.yml]: yeah you're right. looking into making this possible

**Comment by joshiste** [comment]
@joshlong, could you update this pull request?

we now have 2 starters:
codecentric-spring-boot-admin-starter-server
codecentric-spring-boot-admin-starter-client
the old spring-boot-admin-starter-client just relocates to the new one.
and the bom is named spring-boot-admin-depencenies

our docs are now available on a url without referencing a version:
http://codecentric.github.io/spring-boot-admin/current/

You can find a snapshot build here:
https://oss.sonatype.org/content/repositories/snapshots/de/codecentric/

Please let me know if everything is now ok and I'll hit the release button.

**Comment by joshlong** [comment]
Hi @snicoll I've updated the PR per @joshiste and your feedback. WDYT?

**Comment by joshiste** [comment]
Hey @joshlong thanks for updating, 
but imho there are some missing points:

a) wrong url in links to the references. should be [http://codecentric.github.io/spring-boot-admin/**current**/](http://codecentric.github.io/spring-boot-admin/current/)
b) doesn't use our BOM (spring-boot-admin-depencenies)
c) and uses the wrong version? (should be 1.5.7 - which I'd release if someone tells me that bom and starters are now ok)

**Comment by snicoll** [comment]
>  which I'd release if someone tells me that bom and starters are now ok

@joshiste Thanks a lot for looking into this. It doesn't matter if we have a version right now that  implements the proposal as we still are in the process of reviewing it. It's perfectly ok to have the renames on a branch only. And please do not release 1.5.7 as we can't give you that feedback right the way.

**Comment by joshlong** [comment]
I've updated the PR w/ @joshiste 's feedback.

File: initializr-service/src/main/resources/application.yml [hunk 4]
```diff
@@ -1252,6 +1259,26 @@ initializr:
...
+        - name: Codecentric Spring Boot Admin Client
+          id: codecentric-spring-boot-admin-starter-client
+          bom: codecentric-spring-boot-admin
+          description: Registers applications with the Spring Boot Admin. Alternatively, Spring Boot Admin supports Spring Cloud DiscoveryClient implementations like Eureka or Consul.
+          artifactId: codecentric-spring-boot-admin-starter-client
+#          version: 1.5.7
+          groupId: de.codecentric
+          links:
+            - rel: reference
+              href: http://codecentric.github.io/spring-boot-admin/1.5.6/#getting-started
```
Comment (joshiste) [inline initializr-service/src/main/resources/application.yml]: should be http://codecentric.github.io/spring-boot-admin/current/#getting-started

**Comment by pivotal-issuemaster** [comment]
@joshlong Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=spring-io/initializr&pullRequestId=571)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=spring-io/initializr&pullRequestId=571) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@joshlong Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=spring-io/initializr&pullRequestId=571)!

**Pull request closed without merging**